slsReceiverSoftware/src/slsReceiverTCPIPInterface.cpp

```diff
@@ -682,1580 +547,793 @@ int slsReceiverTCPIPInterface::set_roi(sls::ServerInterface2 &socket) {
+	strcpy(retval, impl()->getAdditionalJsonHeader().c_str());
+	FILE_LOG(logDEBUG1) << "additional json header:" << retval;
+	return socket.sendResult(retval);
+}
+
+int slsReceiverTCPIPInterface::set_udp_socket_buffer_size(Interface &socket) {
+	auto index = socket.receive<int64_t>();
+	if (index >= 0) {
+		VerifyIdle(socket);
+		FILE_LOG(logDEBUG1) << "Setting UDP Socket Buffer size: " << index;
+		if (impl()->setUDPSocketBufferSize(index) == FAIL) {
+			throw RuntimeError("Could not create dummy UDP Socket to test buffer size");
 		}
-		// get
-		strcpy(retval, receiver->getAdditionalJsonHeader().c_str());
-		FILE_LOG(logDEBUG1) << "additional json header:" << retval;
 	}
-	 return socket.sendResult(true, ret, retval, MAX_STR_LENGTH, mess);
+	int64_t retval = impl()->getUDPSocketBufferSize();
+	if (index != 0)
+		validate(index, retval, std::string("set udp socket buffer size (No CAP_NET_ADMIN privileges?)"), DEC);
+	FILE_LOG(logDEBUG1) << "UDP Socket Buffer Size:" << retval;
+	return socket.sendResult(retval);
 }
 
-
-
-int slsReceiverTCPIPInterface::get_additional_json_header(sls::ServerInterface2 &socket) {
-	ret = OK;
-    memset(mess, 0, sizeof(mess));
-    char retval[MAX_STR_LENGTH] = {0};
-
-	if(receiver == nullptr){
-		NullObjectError(ret, mess);
-	}
-
-	// base object not null
-	if (ret == OK) {
-		// get
-		strcpy(retval, receiver->getAdditionalJsonHeader().c_str());
-		FILE_LOG(logDEBUG1) << "additional json header:" << retval;
-	}
-	 return socket.sendResult(true, ret, retval, MAX_STR_LENGTH, mess);
-}
-
-
-
-int slsReceiverTCPIPInterface::set_udp_socket_buffer_size(sls::ServerInterface2 &socket) {
-	ret = OK;
-    memset(mess, 0, sizeof(mess));
-    int64_t index = -1;
-    int64_t retval = -1;
-
-	// get args, return if socket crashed, ret is fail if receiver is not null
-	socket.receiveArg(ret, mess, &index, sizeof(index));
-	if(receiver == nullptr){
-		NullObjectError(ret, mess);
-		return FAIL;
-	}
-
-	// base object not null
-	if (ret == OK) {
-		// set
-		if (index >= 0) {
-			// verify if receiver is unlocked and idle
-			if (VerifyLockAndIdle(ret, mess, fnum) == OK) {
-				FILE_LOG(logDEBUG1) << "Setting UDP Socket Buffer size: " << index;
-				if (receiver->setUDPSocketBufferSize(index) == FAIL) {
-                    ret = FAIL;
-                    strcpy(mess, "Could not create dummy UDP Socket to test buffer size\n");
-                    FILE_LOG(logERROR) << mess;
-                }
-			}
-		}
-		// get
-        retval = receiver->getUDPSocketBufferSize();
-        if (index != 0)
-        	validate(index, retval, std::string("set udp socket buffer size (No CAP_NET_ADMIN privileges?)"), DEC);
-        FILE_LOG(logDEBUG1) << "UDP Socket Buffer Size:" << retval;
-	}
-	return socket.sendResult(true, ret, &retval, sizeof(retval), mess);
+int slsReceiverTCPIPInterface::get_real_udp_socket_buffer_size(Interface &socket){
+	return socket.sendResult(impl()->getActualUDPSocketBufferSize());
```

Comment:
minor. logdebug print?

Reply:
But I like one liners =) 
Anyway fixed it. Later we can think about how many debug messages we need....